src/ios-deploy/ios-deploy.m

```diff
@@ -1714,30 +1729,84 @@ void handle_device(AMDeviceRef device) {
         check_error(AMDeviceStopSession(device));
         check_error(AMDeviceDisconnect(device));
 
-        CFStringRef keys[] = { CFSTR("PackageType") };
-        CFStringRef values[] = { CFSTR("Developer") };
-        CFDictionaryRef options = CFDictionaryCreate(NULL, (const void **)&keys, (const void **)&values, 1, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
-
-        //assert(AMDeviceTransferApplication(afcFd, path, NULL, transfer_callback, NULL) == 0);
-        check_error(AMDeviceSecureTransferPath(0, device, url, options, transfer_callback, 0));
-        close(*afcFd);
-
+        bool incremental_install = app_deltas != NULL;
 
+        CFDictionaryRef options;
+        if (!incremental_install) {
+          CFStringRef keys[] = { CFSTR("PackageType") };
+          CFStringRef values[] = { CFSTR("Developer") };
+          options = CFDictionaryCreate(NULL, (const void **)&keys, (const void **)&values, 1, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
+          //assert(AMDeviceTransferApplication(afcFd, path, NULL, transfer_callback, NULL) == 0);
+          check_error(AMDeviceSecureTransferPath(0, device, url, options, transfer_callback, 0));
+          close(*afcFd);
+        }
 
         AMDeviceConnect(device);
         assert(AMDeviceIsPaired(device));
         check_error(AMDeviceValidatePairing(device));
         check_error(AMDeviceStartSession(device));
 
-        // // NOTE: the secure version doesn't seem to require us to start the installation_proxy service
-        // // Although I can't find it right now, I in some code that the first param of AMDeviceSecureInstallApplication was a "dontStartInstallProxy"
-        // // implying this is done for us by iOS already
+        if (!incremental_install) {
+          // // NOTE: the secure version doesn't seem to require us to start the installation_proxy service
```

Comment:
These comments are 6 years old and the "non secure" codepaths haven't been exercised since then. https://github.com/ios-control/ios-deploy/pull/48

Since this PR is going to modify the blame for these comments(making it harder to get context), they would be better off removed. You also left some of the code that was originally commented out about 40 lines below(Adding another note on that commented-out code to be deleted `// close(installFd);`).

Reply:
I left it in there since I assume there was a reason for it to have been left in the first place. I was not aware it had been 6 years so I have removed the commented out code unless @shazron has any objections.